src/util/cmirdmautils.h

```diff
@@ -13,28 +13,28 @@ typedef struct ncpystruct{
 #endif
 
   const void *srcPtr;
-  int srcPe;
   char *srcLayerInfo;
-  int srcLayerSize;
   char *srcAck;
-  int srcAckSize;
-  int srcSize;
-  unsigned short int srcRegMode;
-  unsigned short int srcDeregMode;
-  unsigned short int isSrcRegistered;
   const void *srcRef;
+  int srcPe;
+  int srcSize;
+  short int srcLayerSize;
+  short int srcAckSize;
+  unsigned char srcRegMode;
+  unsigned char srcDeregMode;
+  unsigned char isSrcRegistered;
```

Comment:
Since we've moved to all C++ now, we can change `isSrcRegistered` and `isDestRegistered` to bools (for easier usage, but that can happen in a separate patch as well since that'll need machine layer changes).

Pes and size variables can be made unsigned too. (doesn't include rootNode though).

Reply:
PEs are signed ints everywhere else in the runtime, and size variables are also signed ints in most of the communication layers AFAIK, so making them unsigned here won't really get us anything. Changing to bool would be good for those that can be